tests/test_backends.py

```diff
@@ -277,3 +277,14 @@ def test_backend_mps_n_mode_perm_decomp():
     check_output_distribution(backend, BasicState("|1,0,1>"),
                               {BasicState("|0,2,0>"): 0.5,
                                BasicState("|0,0,2>"): 0.5})
+
+
+def test_probampli_iterator_cache():
+    backends = [SLOSBackend(), MPSBackend(), NaiveBackend()]
```

Comment:
We usually do a parametrized auto test with `@pytest.mark.parametrize` instead of looping over different values (see test_backend_cnot for an example).

You can `@pytest.mark.parametrize(backend_type, (SLOSBackend, MPSBackend, NaiveBackend))` and then call
`b = backend_type()` to instanciate

Reply:
ok